app/labor/rate_limit_checker.rb

```diff
@@ -40,4 +43,14 @@ def limit_by_email_recipient_address(address)
   def ping_admins
     RateLimitCheckerJob.perform_later(user.id, action)
   end
+
+  private
+
+  def user_today_follow_count
+    now = Time.zone.now
+    day_start = now.beginning_of_day
+    day_end = now.end_of_day
+
+    user.follows.where(created_at: (day_start..day_end)).size
```

Comment:
Given this statement is going to be run a lot it would probably be a good idea to ensure created_at is indexed.  Would you mind adding an index to `created_at` on the users table?

Reply:
I still don't know why Rails doesn't generate indexes on `created_at` by default :D

Reply:
This could also be simplified by only checking up until now, correct? There's no need to check in the future, we can't create follows in the future. So the query could be:

```ruby
user.follows.where('created_at < ?', Time.current).size
```

let me know if i'm missing something here

Reply:
Good call, also I had to google to make sure Rails didnt auto create indexes bc I honestly thought they would have done it automatically as well.

Reply:
My bad, since the point is "daily" usage, you still need the lower limit, so the query would be:

```ruby
now = Time.current
user.follows.where(created_at: (now.beginning_of_day..now)).size
```

Reply:
We could also avoid this query in most cases by putting up a guard clause related to `user.following_users_count` since that is a counter cache on how many users the person is already following. If it is fewer than 500 total, no further action is ever required.